src/tribol/mesh/MfemData.hpp

```diff
@@ -0,0 +1,1390 @@
+   * 
+   * @return const mfem::GridFunction& 
+   */
+  const mfem::GridFunction& GetRedecompGap() const
+  {
+    return redecomp_gap_;
+  }
+
+  /**
+   * @brief Get the gap grid function on the (coarse) submesh
+   * 
+   * @return mfem::ParGridFunction 
+   */
+  mfem::ParGridFunction GetSubmeshGap() const
+  {
+    return GetPressureTransfer().RedecompToSubmesh(redecomp_gap_);
+  }
+
+  /**
+   * @brief Get the submesh to redecomp pressure transfer object
+   * 
+   * @return const SubmeshRedecompTransfer& 
+   */
+  const SubmeshRedecompTransfer& GetPressureTransfer() const
+  {
+    return GetUpdateData().pressure_xfer_;
+  }
+
+  /**
+   * @brief Get the finite element space on the (coarse) submesh
+   * 
+   * @return const mfem::ParFiniteElementSpace& 
+   */
+  const mfem::ParFiniteElementSpace& GetSubmeshFESpace() const
+  {
+    return *submesh_pressure_.ParFESpace();
+  }
+
+  /**
+   * @brief Get the finite element space on the refined submesh
+   *
+   * @return const mfem::ParFiniteElementSpace* or nullptr if no refined submesh
+   */
+  const mfem::ParFiniteElementSpace* GetLORSubmeshFESpace() const
+  {
+    return lor_xfer_ ? lor_xfer_->GetLORGridFn().ParFESpace() : nullptr;
+  }
+
+private:
+  /**
+   * @brief Creates and stores data that changes when the redecomp mesh is
+   * updated
+   */
+  struct UpdateData
+  {
+    /**
+     * @brief Construct a new UpdateData object
+     *
+     * @param submesh_fes Pressure finite element space on the (coarse) submesh
+     * @param lor_xfer (Optional) Low-order grid function, refined mesh transfer
+     * object
+     * @param redecomp Updated (refined) redecomp mesh
+     */
+    UpdateData(
+      mfem::ParFiniteElementSpace& submesh_fes,
+      SubmeshLORTransfer* lor_xfer,
+      redecomp::RedecompMesh& redecomp
+    );
+
+    /**
+     * @brief Submesh to redecomp field transfer object
+     */
+    SubmeshRedecompTransfer pressure_xfer_;
+  };
+
+  /**
+   * @brief Get the UpdateData object
+   * 
+   * @return UpdateData& 
+   */
+  UpdateData& GetUpdateData();
+
+  /**
+   * @brief Get the UpdateData object
+   * 
+   * @return const UpdateData& 
+   */
+  const UpdateData& GetUpdateData() const;
+
+  /**
+   * @brief Pressure grid function on the (coarse) submesh
+   */
+  mfem::ParGridFunction submesh_pressure_;
+
+  /**
+   * @brief Pressure grid function and transfer operators
+   */
+  PressureField pressure_;
+
+  /**
+   * @brief Contains low-order refined submesh transfer operators if low-order
+   * refinement is being used; nullptr otherwise
+   */
+  std::unique_ptr<SubmeshLORTransfer> lor_xfer_;
+
+  /**
+   * @brief Gap grid function on the redecomp mesh
+   */
+  mfem::GridFunction redecomp_gap_;
+
+  /**
+   * @brief UpdateData object created upon call to UpdateSubmeshData()
+   */
+  std::unique_ptr<UpdateData> update_data_;
+};
+
+/**
+ * @brief Simplifies transfer of matrix data between MFEM and Tribol
```

Comment:
Can you clarify here what matrix data needs to be transferred? Is this Jacobian data? MfemMatrixData is a little ambiguous.

Reply:
Updated to `MfemJacobianData`